api/renter.go

```diff
@@ -350,49 +387,141 @@ func (api *API) renterDeleteHandler(w http.ResponseWriter, req *http.Request, ps
 
 // renterDownloadHandler handles the API call to download a file.
 func (api *API) renterDownloadHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	destination := req.FormValue("destination")
-	// Check that the destination path is absolute.
-	if !filepath.IsAbs(destination) {
-		WriteError(w, Error{"destination must be an absolute path"}, http.StatusBadRequest)
+	p, errmsg := api.parseAndValidateDownloadParameters(w, req, ps)
```

Comment:
Per my comment below, I think it would be good to separate the parsing and the validating. Parsing everything in one place makes sense to me, but I think that validation should happen alongside the use (makes it easier during both writing, review, and future modification to verify that validation + use match up correctly, b/c the code is all in the same place)